### ISSUE TITLE
CERTIFICATE_VERIFY_FAILED

### ISSUE THREAD

**Issue by mathmaniac43** [git-touch/git-touch#102]
Hello,

I am trying to set up this app on my Motorola Android 9.0 phone. I want to connect it to my locally-hosted GiTea instance, which I have generated my own SSL certificates up to use HTTPS with. I have installed the root certificate onto my phone, and it works for my web browser and other services.

When I plug in the GiTea access token, I get the following error:

```
Something bad happens: HandshakeException: Handshake error in client
(OS Error:

CERTIFICATE_VERIFY_FAILED: unable to get local issuer certificate(handshake.cc:354))
```

Is something preventing the app from using the system's user-installed SSL certificates?

Thanks!

**Comment by pd4d10** [comment]
Hi, @mathmaniac43 , thanks for the report!

Does you self-hosted site work fine in the browser?

**Comment by mathmaniac43** [comment]
Yes it does. After I installed the public root cert I generated on my phone, the browsers (Firefox and Google Chrome) are happy to open the site.

**Comment by pd4d10** [comment]
OK. It looks related to Dart/Flutter framework, not the app itself.

Found some similar issues at Flutter repo: https://github.com/flutter/flutter/issues/50699 https://github.com/flutter/flutter/issues/23045

**Comment by mathmaniac43** [comment]
I see, thank you for finding and sharing that!

**Comment by kakolisgay** [comment]
Potential fix [flutter/flutter#50699#issuecomment-653855688](https://github.com/flutter/flutter/issues/50699#issuecomment-653855688)

**Comment by HarrPerson** [comment]
I have the same problem with GitTouch and gogs.
Self signed root CA is installed as user certificate. Chrome works fine.
